hnn_core/cell.py

```diff
@@ -363,9 +440,18 @@ def _set_biophysics(self, sections):
                 for attr, val in p_mech.items():
                     if hasattr(val, '__call__'):
                         sec.push()
+                        seg_xs, seg_vals = list(), list()
                         for seg in sec:
-                            setattr(seg, attr, val(h.distance(seg.x)))
+                            seg_xs.append(seg.x)
+                            seg_vals.append(val(h.distance(seg.x)))
+                            setattr(seg, attr, seg_vals[-1])
                         h.pop_section()
+                        p_mech[attr] = [seg_xs, seg_vals]
```

Comment:
I'm assuming the purpose of this is to overwrite the callable stored in `p_mech`? If so, this will require calling `cell.build()` any time you need to write write a cells attached to a `Network` instance, which will in turn imply that we're going to instantiate NEURON objects from within `Network`. I'm just thinking out loud here, but will that cause any unintended issues with the NEURON network? In the past, we've maintained the convention that NEURON objects are only instantiated from `NetworkBuilder` in order to silo how, when, and where such objects are created.

Reply:
Yes, calling `cell.build()` outside the `NetworkBuilder` does throw an error later when Network is simulated. For resolving it, I first made a copy of the cell and called `build()` on the copy instead of the original.

Reply:
Do we know if the NEURON objects are getting properly destroyed and de-referenced after the cell copy is destroyed by the garbage collector? The only reason I'm bringing this up is because great care needs to be taken when handling NEURON network objects which are generally global and have in the past caused caused segmentation faults that are hard to diagnose and/or make tests for.

Is there a way around this that allows us to maintain our convention of only building NEURON objects from within `NetworkBuilder`? CC'ing @jasmainak and @ntolley for their input.

Reply:
yeah I agree we need to be careful here. See: https://github.com/jonescompneurolab/hnn-core/blob/c81e5b87fbcc13bd4c6e5a403e838ca63ea16760/hnn_core/network_builder.py#L658 

The trade-off is between doing our computation ... which can be error-prone if we don't get the positions of the segments exactly right. Neuron does some mumbo-jumbo here:

https://github.com/jonescompneurolab/hnn-core/blob/c81e5b87fbcc13bd4c6e5a403e838ca63ea16760/hnn_core/cell.py#L431-L434

vs

potentially slowing up or causing segmentation faults in all our simulations that have run the `save` method

Reply:
The fact that the segment coordinates prior to NEURON instantiation aren't their final values (and thus need to be re-set using length modifiers) is annoying. Maybe we should rip the bandaid off and fix this, which I think would also nullify the need to build the NEURON cells in order to write the network.

Reply:
I'm proposing the latter, which I think would allow us to get rid of the length parameter. While this may introduce small numerical differences compared to what we currently have, it's a much more sustainable solution.

Reply:
what would be concrete steps for @raj1701 ?

1. Raise warning if length is defined in `param` file? It should say that it has no effect. Same with network attribute : they should not be modifiable
2. Update the coordinates here to match the default length parameters: https://github.com/jonescompneurolab/hnn-core/blob/c81e5b87fbcc13bd4c6e5a403e838ca63ea16760/hnn_core/cells_default.py#L69-L77
3. Remove `h.define_shape` line

Anything else I missed? Does this sound doable @raj1701 ? Any inputs @ntolley ?

Reply:
I think there are a quite a few params in the param files that are silently ignored already, right? Since most of this stuff is already specified in [`params_default.py`](https://github.com/jonescompneurolab/hnn-core/blob/c81e5b87fbcc13bd4c6e5a403e838ca63ea16760/hnn_core/params_default.py#L286), is the param file warning necessary?

We should also remove the lines where the NEURON length and radius attributes are set (see [here](https://github.com/jonescompneurolab/hnn-core/blob/c81e5b87fbcc13bd4c6e5a403e838ca63ea16760/hnn_core/cell.py#L409C1-L412C43)).